File: solutions/relocation/relocation.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

lc = 0;
D = null;

parseLine = line => {
  return line.split(' ').map(s => parseInt(s));
};

rl.on('line', line => {
  if (lc == 0) {
  } else if (lc == 1) {
    D = parseLine(line);
  } else {
    q = parseLine(line);
    if (q[0] == 1) {
      D[q[1] - 1] = q[2]
    } else {
      console.log(Math.abs(
        D[q[1] - 1] - D[q[2] - 1]
      ));
    }
  }
  lc++;
});

File: solutions/relocation/relocation.rb
def rl
  STDIN.readline.split.map &:to_i
end

N, Q = rl
D = rl
Q.times do
  type, a, b = rl
  if type == 1
    D[a - 1] = b
  else
    puts (D[a - 1] - D[b - 1]).abs
  end
end
